src/validation-engine.js

```diff
@@ -67,6 +69,16 @@ class ValidationEngine {
 
     if (shape.deactivated) return false
 
+    // check recursion depth: how many times have we already checked this focusNode against this shape?
+    let id = JSON.stringify([focusNode, shape.shapeNode])
+    let recursionDepth = this.checkedNodes[id] === undefined ? 0 : this.checkedNodes[id]
```

Comment:
```suggestion
    const id = JSON.stringify([focusNode, shape.shapeNode])
    const recursionDepth = this.checkedNodes[id] === undefined ? 0 : this.checkedNodes[id]
```

Reply:
done by latest commit. Also I have renamed the variables to reflect what they are for.

Setting `maxNodeChecks` to `0` in the options disables the recursion protection now. My tests showed, that a value of `300` for `maxNodeChecks` still works without the runtime throwing `maximum call stack size exceeded`.

Reply:
Super! Thanks for you effort.